src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,81 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo }
         };
+  }
+
+  handleSettingsClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+
+ 
+    let indexOf = this.props.svgOptions.findIndex(this.props.value);
+
+    if (indexOf < 0) {
+      this.props.svgOptions = this.props.svgOptions.concat(this.props.value);
+    } else {
+      this.props.svgOptions = this.props.svgOptions.slice(0,indexOf).concat(this.props.svgOptions.slice(indexOf + 1))
```

Comment:
never a good idea to overwrite a prop value, (mutation)

Reply:
I am no longer assigning direct like that. Its fixed